db/db_impl_compaction_flush.cc

```diff
@@ -1943,7 +1947,7 @@ void DBImpl::SchedulePendingPurge(std::string fname, std::string dir_to_sync,
 void DBImpl::BGWorkFlush(void* db) {
   IOSTATS_SET_THREAD_POOL_ID(Env::Priority::HIGH);
   TEST_SYNC_POINT("DBImpl::BGWorkFlush");
-  reinterpret_cast<DBImpl*>(db)->BackgroundCallFlush();
+  reinterpret_cast<DBImpl*>(db)->BackgroundCallFlush(Env::Priority::HIGH);
```

Comment:
Flush can also be scheduled onto a thread in the low-priority pool. See https://github.com/facebook/rocksdb/blob/master/db/db_impl_compaction_flush.cc#L1791

Reply:
You may need to add a new member variable to `BGFlushArg` and set it in `MaybeScheduleFlushOrCompaction`. Then `BackgroundCallFlush` can read it while popping elements from the queue.